src/BuildScriptGenerator/DotNetCore/DotnetCorePlatform.cs

```diff
@@ -61,43 +66,273 @@ public BuildScriptSnippet GenerateBashBuildScriptSnippet(BuildScriptGeneratorCon
-            {
-                ProjectFile = projectFile,
-                PublishDirectory = publishDir,
-                BuildProperties = buildProperties,
-                BenvArgs = $"dotnet={context.DotNetCoreVersion}",
-                DirectoriesToExcludeFromCopyToIntermediateDir = GetDirectoriesToExcludeFromCopyToIntermediateDir(
-                    context),
-                PreBuildCommand = preBuildCommand,
-                PostBuildCommand = postBuildCommand,
-                ManifestFileName = FilePaths.BuildManifestFileName,
-                ManifestDir = context.ManifestDir,
-                ZipAllOutput = zipAllOutput,
-                Configuration = GetBuildConfiguration(),
+            var scriptBuilder = new StringBuilder();
+            scriptBuilder
+                .AppendLine("#!/bin/bash")
+                .AppendLine("set -e")
+                .AppendLine();
+
+            var sourceDir = _buildOptions.SourceDir;
+            var destinationDir = _buildOptions.DestinationDir;
+            var userSuppliedDestinationDir = !string.IsNullOrEmpty(_buildOptions.DestinationDir);
+            var zipAllOutput = ShouldZipAllOutput();
+
+            AddScriptToCopyToIntermediateDirectory();
+
+            AddScriptToSetupSourceAndDestinationDirectories();
+
+            scriptBuilder.AppendBenvCommand($"dotnet={context.DotNetCoreVersion}");
+
+            AddScriptToRunPreBuildCommand();
+
+            scriptBuilder
+                .AppendLine("echo")
+                .AppendLine("dotnetCoreVersion=$(dotnet --version)")
+                .AppendLine("echo \"Using .NET Core SDK Version: $dotnetCoreVersion\"")
+                .AppendLine();
+
+            AddScriptToRestorePackages();
+
+            if (userSuppliedDestinationDir)
+            {
+                if (zipAllOutput)
+                {
+                    AddScriptToZipAllOutput();
+                }
+                else
+                {
+                    AddScriptToPublishOutput();
+
+                    AddScriptToRunPostBuildCommand();
+                }
+            }
+            else
+            {
+                AddScriptToBuildProject();
+
+                AddScriptToRunPostBuildCommand();
+            }
+
+            SetStartupFileNameInfoInManifestFile(context, projectFile, buildProperties);
+
+            AddScriptToCreateManifestFile();
+
+            scriptBuilder.AppendLine("echo Done.");
+
+            return new BuildScriptSnippet
+            {
+                BashBuildScriptSnippet = scriptBuilder.ToString(),
+                IsFullScript = true,
             };
-            var script = TemplateHelper.Render(
-                TemplateHelper.TemplateResource.DotNetCoreSnippet,
-                templateProperties,
-                _logger);
-            return new BuildScriptSnippet { BashBuildScriptSnippet = script, IsFullScript = true };
+
+            void AddScriptToCopyToIntermediateDirectory()
```

Comment:
having those inline methods make things harder to read IMO. How about we add this to a separate builder class with fluid API, without the `Add` in each method? Or as extension methods within a "dotnet" child namespace?
```
buildScript
     .RunPreBuildCommand()
     .ZipAllOutput()
     .RunPostBuildCommand()
     .CreateManifestFile()
...
```